Assets/Plugins/iOS/Shopify/Cart+ApplePay.mm

```diff
@@ -0,0 +1,97 @@
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "Cart+ApplePay.h"
+#import "ProductName-Swift.h"
+
+ApplePayEventDispatcher *dispatcher;
+
+bool _CanCheckoutWithApplePay() {
+    return [PaymentSession canMakePayments];
+}
+
+bool _CanShowApplePaySetup() {
+    return [PaymentSession canShowSetup];
+}
+
+void _ShowApplePaySetup() {
+    [PaymentSession showSetup];
+}
+
+bool _CreateApplePaySession(const char *merchantID, const char *countryCode, const char *currencyCode, bool requiringShipping, const char *unityDelegateObjectName, const char *serializedSummaryItems, const char *serializedShippingMethods) {
+    
+    NSString *dataKey = @"Items";
+    NSString *itemsString = [NSString stringWithUTF8String:serializedSummaryItems];
+    NSString *shippingString = [NSString stringWithUTF8String:serializedShippingMethods];
+    
+    /// Parse Summary Items
+    NSError *error = nil;
+    NSDictionary *summaryItemsJson = [NSJSONSerialization JSONObjectWithData:[itemsString dataUsingEncoding:NSUTF8StringEncoding]
+                                                                     options:kNilOptions
+                                                                       error:&error];
+    
+    NSArray *summaryItems;
+    if (error == nil && summaryItemsJson != nil && [summaryItemsJson objectForKey:dataKey] != nil) {
+        
+        NSArray *summaryItemsArray = summaryItemsJson[dataKey];
+        summaryItems = [PKPaymentSummaryItem deserializeWithSummaryItems:summaryItemsArray];
+        
+        if (summaryItems == nil) {
+            return false;
+        }
+    } else {
+        return false;
+    }
+
+    /// Parse Shipping Items
+    NSDictionary *shippingItemsJson = [NSJSONSerialization JSONObjectWithData:[shippingString dataUsingEncoding:NSUTF8StringEncoding]
+                                                                      options:kNilOptions
+                                                                        error:&error];
+    
+    NSArray *shippingMethods;
```

Comment:
+ We can create a `PaymentSession` with no `shippingMethods` because we may not have the necessary information to provide shipping information without the user's address